### PR TITLE
BUG: Parsing offset strings with non-zero minutes was incorrect

### LINKED ISSUE #11708: pd.to_datetime("2015-11-18 15:30:00+05:30").tz_localize('UTC').tz_convert('Asia/Kolkata') returns '2015-11-18 16:30:00+0530'
running "pd.to_datetime("2015-11-18 15:30:00+05:30").tz_localize('UTC').tz_convert('Asia/Kolkata')" returns following o/p

Timestamp('2015-11-18 16:30:00+0530', tz='Asia/Kolkata')

It should have returned as Timestamp('2015-11-18 15:30:00+0530', tz='Asia/Kolkata') as no daylight saving is applicable in India.

### PR DESCRIPTION
**Pull request by rockg**
Closes #11708

Minutes needed to be added to the hour offset rather than subtracted.

### CODE REVIEW

File: pandas/tseries/tests/test_tslib.py [hunk 1]
```diff
@@ -722,6 +722,8 @@ def test_parsing_timezone_offsets(self):
             '2013-01-01T08:00:00.000000000+0800',
             '2012-12-31T16:00:00.000000000-0800',
             '12-31-2012 23:00:00-01:00',
+            '2013-01-01 05:45+05:45', #GH11708
+            '2013-01-01 05:30+05:30', #GH11708
```
Comment (jreback) [inline pandas/tseries/tests/test_tslib.py]: this is fine, but we also need a test asserting that these are the correct times when localized (e.g. like the example in the original bug report)

**Comment by jreback** [comment]
this does not close #11481 , correct?

**Comment by rockg** [comment]
Correct, that's a lot uglier.

**Comment by rockg** [comment]
Tests added for to_datetime and Timestamp constructor and are green.

**Comment by jreback** [comment]
thank you sir!

**Pull request merged**